server-spi/src/main/java/org/keycloak/models/SingleEntityCredentialManager.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2022. Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models;
+
+import org.keycloak.credential.CredentialInput;
+import org.keycloak.credential.CredentialModel;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Validates and manages the credentials of a known entity (for example, a user).
+ */
+public interface SingleEntityCredentialManager {
```

Comment:
I would name this one just `CredentialManager` or `SubjectCredentialManager` if you want to be generic enough to support entities other than users.

It is a nitpick, but the name for me is too long and too generic.

Reply:
I like this suggestion - `SubjectCredentialManager` sounds good to me

Reply:
Renaming has been done in b37af25bc4dc4a33a70487eb15c813ab298f165a